src/meta/src/hummock/manager/gc.rs

```diff
@@ -48,6 +48,8 @@ pub(crate) struct GcManager {
     store: ObjectStoreRef,
     path_prefix: String,
     use_new_object_prefix_strategy: bool,
+    /// These objects may still be used by backup or time travel.
+    may_delete_object_ids: parking_lot::Mutex<HashSet<HummockSstableObjectId>>,
```

Comment:
Given that the minor gc is triggered directly after `create_version_checkpoint` and we can always get the full stale objects from `create_version_checkpoint` directly. Will it be better if we instead let `create_version_checkpoint` returns the stale objects without introducing the mutext protected hash set?

Reply:
may_delete_object_ids may also be modified when proactively GC time travel objects. See latest commits.

Reply:
Will refactor.